images/archive/importer.go

```diff
@@ -186,15 +187,24 @@ func ImportIndex(ctx context.Context, store content.Store, reader io.Reader, opt
 			return ocispec.Descriptor{}, errors.Wrap(err, "write docker manifest")
 		}
 
-		platforms, err := images.Platforms(ctx, store, desc)
+		imgPlatforms, err := images.Platforms(ctx, store, desc)
 		if err != nil {
 			return ocispec.Descriptor{}, errors.Wrap(err, "unable to resolve platform")
 		}
-		if len(platforms) > 0 {
+
+		if len(imgPlatforms) == 0 || (imgPlatforms[0].OS == "windows" && imgPlatforms[0].OSVersion == "") {
+			// If the image we've just imported doesn't have a platform, or it's a Windows
+			// image without the OSVersion set, we could just assume it matches this host's
+			// platform. Without this, the children labels might not be set on the image
+			// content, leading to it being garbage collected, breaking the image.
+			// See: https://github.com/containerd/containerd/issues/5690
+			platform := platforms.DefaultSpec()
+			desc.Platform = &platform
```

Comment:
Is this assuming that `platform.OS` is `"windows"` as well?

Reply:
yes, ``platform.OS`` will be ``windows`` on Windows, since ``DefaultSpec`` will get that value from ``runtime.GOOS``: https://github.com/containerd/containerd/blob/ff75f7d4899cdaa019be4cd904fc5c4ab9748b09/platforms/defaults_windows.go#L31

I've actually noticed that the platforms from ``imgPlatforms`` will always have an empty string ``OSVersion`` because ``platforms.Normalize`` is called here https://github.com/containerd/containerd/blob/ff75f7d4899cdaa019be4cd904fc5c4ab9748b09/images/image.go#L280 , which sets it to empty string (https://github.com/containerd/containerd/blob/ff75f7d4899cdaa019be4cd904fc5c4ab9748b09/platforms/platforms.go#L257). Interesting, will have to check where else we call ``platforms.Normalize``.

Anyways, keep in mind that this PR also contains this PR: https://github.com/containerd/containerd/pull/5916, it was needed in order for the CIs to pass.